value.go

```diff
@@ -785,37 +798,53 @@ func (vlog *valueLog) deleteLogFile(lf *logFile) error {
 	if err := lf.fd.Close(); err != nil {
 		return err
 	}
-	return os.Remove(path)
+	y.AssertTrue(lf.path == vlog.fpath(lf.fid, lf.fileType))
+	return os.Remove(lf.path)
 }
 
 func (vlog *valueLog) dropAll() (int, error) {
-	// If db is opened in InMemory mode, we don't need to do anything since there are no vlog files.
+	// If db is opened in InMemory mode, we don't need to do anything since there are no log files.
 	if vlog.db.opt.InMemory {
 		return 0, nil
 	}
 	// We don't want to block dropAll on any pending transactions. So, don't worry about iterator
 	// count.
 	var count int
-	deleteAll := func() error {
-		vlog.filesLock.Lock()
-		defer vlog.filesLock.Unlock()
-		for _, lf := range vlog.filesMap {
+	deleteAll := func(lw *logWrapper) error {
+		lw.filesLock.Lock()
+		defer lw.filesLock.Unlock()
+		for _, lf := range lw.filesMap {
 			if err := vlog.deleteLogFile(lf); err != nil {
 				return err
 			}
 			count++
 		}
-		vlog.filesMap = make(map[uint32]*logFile)
+		lw.filesMap = make(map[uint32]*logFile)
 		return nil
 	}
-	if err := deleteAll(); err != nil {
+	if err := deleteAll(&vlog.vlog); err != nil {
+		return count, err
+	}
+	if err := deleteAll(&vlog.wal); err != nil {
 		return count, err
 	}
 
-	vlog.db.opt.Infof("Value logs deleted. Creating value log file: 0")
-	if _, err := vlog.createVlogFile(0); err != nil { // Called while writes are stopped.
+	y.AssertTrue(len(vlog.vlog.filesMap) == 0)
+	y.AssertTrue(len(vlog.wal.filesMap) == 0)
+
+	vlog.db.opt.Infof("Value logs deleted. Creating log file: 0")
+	if _, err := vlog.createLogFile(0, walFile); err != nil { // Called while writes are stopped.
```

Comment:
Create the vlog files as well. While closing the DB, if the writeable offset of vlog file is zero, delete the file.